src/session/content/message_row/bubble.rs

```diff
@@ -193,6 +201,22 @@ impl MessageBubble {
             None
         };
 
+        // Handle MessageReply
+        let message_chat = message
+            .forward_info()
+            .and_then(|f| message.chat().session().try_chat(f.chat_id()))
```

Comment:
try_chat() returns option, so in case it's none it should be requested from the tdlib

Reply:
I think you can use `chat()` just fine here. Read the description [here](https://github.com/melix99/telegrand/blob/cf48b4b3183d7be6032151d754daeb6577487d54/src/session/mod.rs#L325).

Reply:
![screenshot](https://user-images.githubusercontent.com/110191114/221384765-f0a396eb-ba7b-4c4e-8bc1-999d29e33a1a.png)

Then I guess there is a bug in the tdlib or the implementation is buggy

Reply:
after a bit of debugging I think it might be a tdlib bug or maybe both tdlib bug and implementation bug

Reply:
Tdlib docs [says](https://docs.rs/tdlib/latest/tdlib/types/struct.MessageForwardInfo.html#structfield.from_chat_id) that the id is 0 if unknown, so I guess this is the reason. Btw, what is this useful for?

Reply:
> Tdlib docs [says](https://docs.rs/tdlib/latest/tdlib/types/struct.MessageForwardInfo.html#structfield.from_chat_id) that the id is 0 if unknown

The id of the forwarded message is also 0 unless it's being forwarded to saved messages or `@replies` chat and it's explanation of the crashes. But that still doesn't explain why replies fail to load in saved messages and in `@replies` chat.

Reply:
Correction, do not load replies to forwarded messages in any chat
![tdesktop1](https://user-images.githubusercontent.com/110191114/221435681-6e2361be-849f-4922-b43a-d8e1fb856889.png)
![telegrand](https://user-images.githubusercontent.com/110191114/221435683-b46bf3b7-e2f8-4501-a349-535adfef132e.png)

Reply:
but maybe I should leave it for the next pr

Reply:
oh, looks like it's only set visible to false

Reply:
> oh, looks like it's only set visible to false

Yep, it seems that all replies are actually invisible, as I wrote in another review.

I think that in general the replies should always be loaded using the id of the actual visible chat, and never use the `from_chat_id` from the forward info.

Reply:
> of the actual visible chat

The thing is that is not always possible (like previously mentioned `@replies` chat) and I believe this is not correct. I think I partially fixed the previous issue. I'll push my changes soon

Reply:
Mh, ok, I still have no idea how `@replies` works. I have to test it out. But for regular chats that was correct, I've tested it via https://t.me/JsonDumpBot. If you forward a message with a reply and the replied message itself into it, you get that the id of `reply_to_message` is indeed the id of the other forwarded message.

Reply:
Mh, ok, I still have no idea how `@replies` works. I have to test it out. But for regular chats that was correct, I've tested it via https://t.me/JsonDumpBot. If you forward a message with a reply and the replied message itself into it, you get that the id of `reply_to_message` is indeed the id of the other forwarded message.